lib/server/services/lhcPeriod/LhcPeriodStatisticsService.js

```diff
@@ -146,9 +162,16 @@ class LhcPeriodStatisticsService {
                     { years },
                 );
             }
+            if (beamTypes) {
+                queryBuilder.literalWhere(
+                    `${PDP_BEAM_TYPES_IN_STATISTICS_ATTRIBUTE.query} IN (:beamTypes)`,
```

Comment:
If this works, it is probably some highly un-optmized query! I suppose that you have to run the subquery for every rows in the table to then run the IN

Reply:
I don't whethere there is better solution (still using sequelize)
1. This attribute must be calculated
2. so filtering by it must be done using subquery in WHERE clause or using HAVING clause
3. Using HAVING clause breaks counting, `Model.findAndCountAll` dosn't work anymore as well as `Model.count`. Simpler, `findAll` works fine, but it doesn't provide counting information.
4. So literal in where clause...